torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -493,6 +522,78 @@ def write_pad():
         return should_pad
 
 
+def run_autoheuristic(
+    mat1,
+    mat2,
+    orig_bench_fn,
+    pad_bench_fn,
+    m_padded_length,
+    k_padded_length,
+    n_padded_length,
+    do_bench,
+):
+    def get_context(mat1, mat2):
+        context = AHContext()
+
+        context.add_feature("m", mat1.shape[0])
+        context.add_feature("k", mat1.shape[1])
+        context.add_feature("n", mat2.shape[1])
+
+        mat1_strides = mat1.stride()
+        mat2_strides = mat2.stride()
+        context.add_feature("mat1_stride_0", mat1_strides[0])
+        context.add_feature("mat1_stride_0", mat1_strides[0])
+        context.add_feature("mat1_stride_1", mat1_strides[1])
+        context.add_feature("mat2_stride_0", mat2_strides[0])
+        context.add_feature("mat2_stride_1", mat2_strides[1])
+
+        context.add_feature("m_padded_length", m_padded_length)
+        context.add_feature("k_padded_length", k_padded_length)
+        context.add_feature("n_padded_length", n_padded_length)
+
+        context.add_feature("mat1_iscontig", mat1.is_contiguous(), is_categorical=True)
+        context.add_feature("mat2_iscontig", mat2.is_contiguous(), is_categorical=True)
+
+        context.add_feature("mat1_align_size", get_alignment_size(mat1))
+        context.add_feature("mat2_align_size", get_alignment_size(mat2))
+
+        return context
+
+    def feedback_fn(choice):
```

Comment:
I don't think I would worry about this for now. If you want, you can expose some data from autoheuristic to query the feedback.

But I think this is a bit of a roundabout way to allow the outside choice to avoid benchmarking twice.

Reply:
I added an option to query collected feedback. I agree that it is not too important right now, but I thought something like this could be helpful to speedup the data collection process.